diagnostic.md

```diff
@@ -6,70 +6,90 @@ Place your responses inside the fenced code-blocks where indicated by comments.
 What are two ways of telling rails where to route a GET request to '/books/4'?
 
 ```rb
-// your response here
+before_action :set_patient, only: [:show]
+def show
+    render json: @patient
+end
+
+def set_patient
+    @patient = Patient.find(params[:id])
+end
+private :set_patient
+
+def show
+  @patient = Patient.find(params[:id])
+
+  render json: @patient
+end
 ```
 
 In what file does the above code go?
 
 ```md
-// your response here
+In the `app/controllers/books_controller.rb`.
 ```
 
 In your own words, what does the command `bundle exec rake db:migrate` do?
 
 ```md
-// your response here
+The command `bundle exec rake db:migrate` creates a path for the developer to
+experiment and change their code.
 ```
 
 What is the difference between `rails g scaffold things` and
 `rails g controller things`?
 
 ```md
-// your response here
+Creating a scaffold creates the controller, the model, and the serializer,
+and it inputs the controller with the CRUD actions. Using the latter command
+just creates the controller, but we have to put in the CRUD actions manually.
 ```
 
 What is the job of the `serializer`?
 
 ```md
-// your response here
+The `serializer`'s job is to
 ```
 
 What do we expect the command `Patient.all` to return?
 
 ```md
-// your response here
+The command `Patient.all` is supposed to return all of the patients that we
+have in our database.
 ```
 
 We use the constant `Patient` in the `PatientsController`, but where is it
 actually defined?
 
 ```md
-// your response here
+The constant `Patient` is actually defined in the `ApplicationController`;
+The `PatientsController` inherits from it.
 ```
 
 Why do our `model` files appear to be lacking so much code, when they're doing
 so much for us?
 
 ```md
-// your response here
+The `model` files contain so little because they are validating what should
+be shown in the database; they also show the relationship between two databases.
```

Comment:
Also The methods making use of the Patient model are defined within the controller.